[build] __ANDROID_API__ version is hardcoded to value incompatible with most of the recent android targets

**WARNING: Please, read this note carefully before submitting a new issue:**

It is important to realise that **this is NOT A SUPPORT FORUM**, this is for reproducible BUGS with raylib ONLY.

There are lots of generous and helpful people ready to help you out on [raylib Discord forum](https://discord.gg/raylib) or [raylib reddit](https://www.reddit.com/r/raylib/).

Remember that asking for support questions here actively takes developer time away from improving raylib.

---

Please, before submitting a new issue verify and check:

 - [x] I tested it on latest raylib version from master branch
 - [x] I checked there is no similar issue already reported
 - [x] My code has no errors or misuse of raylib

### Issue description

raylib is seemed to be hardcoding android api version to 21 which is pretty old and incompatible with build targets above 23 ( which is most of the devices out there ).
The value is specified in LibraryConfigurations.cmake.

### Environment

Build for android on windows.

### Issue Screenshot

C:/Users/serge/source/myproject/build_android/arm64-v8a/_deps/raylib-src/src/external/tinyobj_loader_c.h:751:13: error: '__sF' is unavailable: obsoleted in Android 23
    fprintf(stderr, "TINYOBJ: Error reading file '%s': %s (%d)\n", filename, strerror(errno), errno);
            ^
C:/Users/serge/AppData/Local/Android/Sdk/ndk/25.1.8937393/toolchains/llvm/prebuilt/windows-x86_64/sysroot/usr/include/stdio.h:76:18: note: expanded from macro 'stderr'
#define stderr (&__sF[2])
                 ^
C:/Users/serge/AppData/Local/Android/Sdk/ndk/25.1.8937393/toolchains/llvm/prebuilt/windows-x86_64/sysroot/usr/include/stdio.h:72:13: note: '__sF' has been explicitly marked unavailable here
extern FILE __sF[] __REMOVED_IN(23);
            ^
In file included from C:/Users/serge/source/myproject/build_android/arm64-v8a/_deps/raylib-src/src/rmodels.c:69:
C:/Users/serge/source/myproject/build_android/arm64-v8a/_deps/raylib-src/src/external/tinyobj_loader_c.h:1326:15: error: '__sF' is unavailable: obsoleted in Android 23
      fprintf(stderr, "TINYOBJ: Failed to parse material file '%s': %d\n", filename, ret);

--- a/cmake/LibraryConfigurations.cmake
+++ b/cmake/LibraryConfigurations.cmake
@@ -45,7 +45,6 @@ elseif (${PLATFORM} MATCHES "Android")
     set(GRAPHICS "GRAPHICS_API_OPENGL_ES2")
     set(CMAKE_POSITION_INDEPENDENT_CODE ON)
     list(APPEND raylib_sources ${ANDROID_NDK}/sources/android/native_app_glue/android_native_app_glue.c)
-    add_definitions(-DANDROID -D__ANDROID_API__=21)
     include_directories(${ANDROID_NDK}/sources/android/native_app_glue)
     set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,--exclude-libs,libatomic.a -Wl,--build-id -Wl,-z,noexecstack -Wl,-z,relro -Wl,-z,now -Wl,--warn-shared-textrel -Wl,--fatal-warnings -u ANativeActivity_onCreate -Wl,-undefined,dynamic_lookup")
 

